### ISSUE TITLE
File language not saved back to github after editing in ST2

### ISSUE THREAD

**Issue by minimaldesign** [condemil/gist#57]
I open a Textile formatted gist, modify it, then save it.

When I look at the gist back on github, everything's fine beside for its language now set to Text instead of Textile. Is there any way to preserve the formatting?

I should mention it opens as a Textile file within ST2 just fine. It's just the saving that seems to go wrong.

Thanks!

**Comment by gabe19** [comment]
There is definitely still a language selection issue. My gist (https://gist.github.com/gabe19/f6d0590d77909f28fd3b) keeps some of the files as "Smalltalk" even though I never selected this and the files have the .cs extension. The language will be correctly auto-selected to C# when I edit the file but it doesn't hold.

**Comment by AlexHedley** [comment]
I've had the same issue with some ObjC gists I've been creating, added the .m and it worked but I've got older gists with no file extension that saved fine

[Using Safari Version 8.0.2 (10600.2.5)]

**Comment by risyasin** [comment]
Two problems still exist for Chrome latest (public gists). 

* Unable to use custom extension.  
	When i change file name to a xxx.conf, A javascript sets language option to Apacheconf, even if i have changed it to Nginx. then it ignores completes. syntax highlighting also 

* Language option can not be saved when we edit the gist. 
	When in editing gists. whichever option you change the language. it just ignores & saves as "text".

**Comment by armaneshaghi** [comment]
@risyasin I can confirm the second problem is also the case in Safari, Mac OS X Yosemite.

**Comment by viion** [comment]
I also am noticing this, it doesn't save the language anymore.

**Comment by dwightchurchill** [comment]
Confirming the second issue as well. Mac OS X Yosemite 10.10.4 / Chrome 42.0.2311.90

**Comment by bazzilic** [comment]
Same issue, setting gist to C#, reverts back to Text.
This gist: https://gist.github.com/bazzilic/8c63050c818868f98f1d

**Comment by cajuuh** [comment]
it seems like @bazzilic solved my problem, adding file extension makes gist recognize my code.

thank you.

**Comment by Code-Toolbox** [comment]
I believe this is still an issue; surely we don't need the file extension as the language can be chosen manually?

**Comment by deekej** [comment]
I agree with @Code-Toolbox - this is just a workaround and is not the FIX itself. The issue here is opened more than 2 years and still isn't fixed. It should be prioritized more, imho.

**Comment by cajuuh** [comment]
well, like i said it worked for **me**, i agree the issue still there for a long time and sure should have more priorization, for sure this workaround won't work again in another project nor another gist.

**Comment by deekej** [comment]
Well that's the point exactly - I'm not entirely sure, if this workaround will work for other file types and for how long... This makes the gist kind of unattractive for me to use at the moment. And I will soon need something for code snippets.

Is there any way how to raise this issue more directly to the devel team? I don't think fix should be so problematic that somebody could not look into it.

**Comment by viion** [comment]
I contacted GitHub and they said it is currently working as expected, so here is the behavior:

- New Gist + No Filename + Language Set = Saved with language
- New Gist + Filename (no extension) + Language Set = Defaults to "Text"
- New Gist + Filename (with extension) + Language Set = Saved with language

I believe my initial thought was because I normally don't create a file name, only a title. I've never really noticed it. Then one day I decided to put in a file name so my gists were named correctly and I started noticing it not saving the language and this caused me some confusion.

To save language, currently you must either not enter a filename or enter one with the extension.

**Comment by deekej** [comment]
Well, but that's just really confusing concept, IMHO, because I do want to make files with only names (without the filename extension) and assign the programming language to it.

It's because I can sometimes shares smalls scripts, which can be copied to /usr/bin/, /usr/share/bin/, etc. on Linux and I don't want anybody to need to rename it additionally.

I would vote for this functionality to be changed...

**Comment by pixelbacon** [comment]
Agreed. If there's a language select, then I extensions.
More so, we don't want amateur programmers just randomly including snippets just
by file name into their own code by an easy search result.
GitHub, Use one or the other, not both

Sent using CloudMagic [https://cloudmagic.com/k/d/mailapp?ct=pi&cv=6.3.16&pv=8.1]
On Wed, May 20, 2015 at 10:10 AM, condemil/Gist
<reply@reply.github.com>
wrote:
Well, but that's just really confusing concept, IMHO, because I do want to make
files with only names (without the filename extension) and assign the
programming language to it.

It's because I can sometimes shares smalls scripts, which can be copied to
/usr/bin/, /usr/share/bin/, etc. on Linux and I don't want anybody to need to
rename it additionally.

I would vote for this functionality to be changed...

—
Reply to this email directly or view it on GitHub
[https://github.com/condemil/Gist/issues/57#issuecomment-103921348] .[https://github.com/notifications/beacon/ABkODP-E2GgARcRIbzrS6McACVWqMY1Bks5oLJtggaJpZM4AQlpE.gif]

**Comment by frob** [comment]
I do Drupal development. Drupal uses custom file extensions for .module and .info and .install

Github thinks that these files are just text files no matter how many times I change this in the ui. The UI setting should override whatever Github thinks based on the file extension. Not every system/framework/engine uses standard file extensions.

**Comment by geosmart** [comment]
thanks BlakeJenningsJohnson ,just add the sufffix of the format,everything is ok

**Comment by frob** [comment]
@geosmart The issue is now that a file extension shouldn't be required and if the user selects the language github should use that extension regardless of what the file extension is.

**Comment by stampycode** [comment]
Is this ever going to be fixed?
I actually keep my custom git hooks in a Gist, and they require specific filenames, which can be in several languages, and the files don't have file extensions. I can't add extensions, as this would break the hooks when I pull the files.

So my Gist currently doesn't have any highlighting, despite me saving the Gist with a specific language from the dropdown.

Seriously, this isn't a difficult problem. BTW, I'm an enterprise customer - so I'm paying for this.
